num/src/bigint.rs

```diff
@@ -848,6 +848,54 @@ macro_rules! construct_bigint {
             }
         }
 
+        impl core::fmt::UpperHex for $name {
+            fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> Result<(), core::fmt::Error> {
+                if f.alternate() {
+                    write!(f, "0x")?;
```

Comment:
If I am not wrong, rust uses `0X` prefix for upper hex - but I personally always prefer `0x`

Reply:
> If I am not wrong, rust uses `0X` prefix for upper hex

I think you're wrong on this one. The [documentation](https://doc.rust-lang.org/std/fmt/trait.UpperHex.html) uses `0x` and [code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f46f0c45d753f4c3a8fb04f8d0ee87de) aswell